src/mnode/mnode-pastel.cpp

```diff
@@ -1297,28 +1297,27 @@ CAmount CArtTradeTicket::GetExtraOutputs(std::vector<CTxOut>& outputs) const
     }
     
     std::string strRoyaltyAddress;
-    if (artRegTicket->nRoyalty > 0)
-    {
-        nRoyaltyAmount = nPriceAmount*artRegTicket->nRoyalty/100;
-        nPriceAmount -= nRoyaltyAmount;
-        
-        //HERE -> Search if there is RoyaltyChange ticket
-    
+    if (artRegTicket->nRoyalty > 0) {
+      nRoyaltyAmount = nPriceAmount * artRegTicket->nRoyalty / 100;
+
+      strRoyaltyAddress = CArtRoyaltyTicket::FindAddressByArtTnxId(artRegTicket->GetTxId());
+      if (strRoyaltyAddress.empty()) {
         CPastelIDRegTicket artistPastelIDticket;
         if (!CPastelIDRegTicket::FindTicketInDb(artRegTicket->pastelIDs[CArtRegTicket::artistsign], artistPastelIDticket))
-            throw std::runtime_error(strprintf(
-                    "The Artist PastelID [%s] from Art Registration ticket with this txid [%s] is not in the blockchain or is invalid",
-                    artRegTicket->pastelIDs[CArtRegTicket::artistsign], artRegTicket->GetTxId()));
+          throw std::runtime_error(strprintf(
+                  "The Artist PastelID [%s] from Art Registration ticket with this txid [%s] is not in the blockchain or is invalid",
+                  artRegTicket->pastelIDs[CArtRegTicket::artistsign], artRegTicket->GetTxId()));
         
         strRoyaltyAddress = artistPastelIDticket.address;
+      }
     }
     
-    if (!artRegTicket->strGreenAddress.empty())
-    {
-        nGreenNFTAmount = nPriceAmount*2/100;
-        nPriceAmount -= nGreenNFTAmount;
+    if (!artRegTicket->strGreenAddress.empty()) {
+      nGreenNFTAmount = nPriceAmount * 2 / 100;
```

Comment:
Does it make sense to define global const in mnode-consts.h?
I see that it is used at least twice in the code.
constexpr uint16_t GREEN_NFT_FEE_PERCENT = 2;

Reply:
agree. will update